syntax = "proto3";

package recommendation;

service RecommendationService {
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
  rpc GetRecommendations(GetRecommendationsRequest) returns (GetRecommendationsResponse);
}

message GetRecommendationsRequest {
  string user_id = 1;
}

message GetRecommendationsResponse {
  repeated Recommendation recommendations = 1;
}

message Recommendation {
    string id = 1;
    string title = 2;
    string author = 3;
    string description = 4;
    string copies = 5;
    string copies_available = 6;
    string category = 7;
    string image_url = 8;
    float price = 9;
    repeated string tags = 10;
}

message HealthCheckRequest {
}

message HealthCheckResponse {
  string status = 1;
}